### DISCUSSION TITLE
Rust/Cargo support is undocumented

### DISCUSSION THREAD

**Discussion by rogusdev** [renovatebot/renovate#17081] [Request Help]
Per this issue, it seems that rust/cargo has support as of 2019:
https://github.com/renovatebot/renovate/issues/1870
and release: https://github.com/renovatebot/renovate/releases/tag/14.37.0

But the docs do not list it in language support:
https://docs.renovatebot.com/ruby/

Is this listed somewhere else? Should the docs be updated? Or is rust not actually supported?

I do see that there is an outstanding concern around the version of rust used:
https://github.com/renovatebot/renovate/issues/6365

With some docs updates planned:
https://github.com/renovatebot/renovate/pull/15791

I'd love to confirm in the docs if renovatebot is useable for rust, and if so, how.

Thanks!

**Comment by viceice** [comment]
https://docs.renovatebot.com/modules/manager/cargo/

**Reply by viceice** [reply]
it's not ruby, it's Rust

**Reply by rogusdev** [reply]
Correct, I just linked to a language that is in the list, since rust is not there

**Reply by viceice** [reply]
not all languages are in that list.

**Reply by rogusdev** [reply]
Interesting, thanks for that link. I did not realize that there was a separate place for language support. When I go looking, I see "Language Support" immediately on the left in the ToC. The page you provided comes from a fairly buried point in the tree: https://docs.renovatebot.com/modules/manager/ > renovate modules > managers. I guess I'm suggesting that the language support list should be expanded to cover everything on that "Supported Managers" list, or that some other change should be made to make that list more discoverable, or connect them better.

**Reply by rogusdev** [reply]
Perhaps said another way: @viceice why are not all languages in that list? That is where I went looking because it is so immediately visible, and did not think I should also look somewhere else.

**Reply by epage** [reply]
I also was confused by the lack of Rust in the language list.

**Comment by epage** [comment]
Some aspects that I feel could be clarified for rust/cargo support in documentation
- With "major", "minor", and "update", how are "0.x.y" and "0.0.x" versions treated?
  - Cargo always treats "x" as the major and "y" as the minor
- What are the `matchDepTypes`?   Digging into the code, it seems they are the sections `dependencies`, `dev-dependencies`, and `build-dependencies`, but what about target-specific dependencies?  If they are all separate, how do I match all dev-dependencies without enumerating every possible target?

Maybe a quick way to describe what I'm trying to do:
- For all dev dependencies, `bump` compatible versions in a single PR
- For all dependencies, `bump` incompatible versions only in individual PRs

The problem is how do I define compatible versions within renovatebot?  How do I get all dev dependencies?

**Reply by viceice** [reply]
for renovate the major is the major. there is no special handling for zero major or minor. you can block automerge via package rules and match current version.

**Reply by epage** [reply]
Thanks for the clarification that they are the absolute and not relative fields and the pointer on current version.
